File: coresuite-dotnet-core-console-cs/generator/MailingLabels.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;

namespace coresuite_dotnet_core_console_cs.generator
{
    public class MailingLabels
    {
        //////////////////////////////////////////////////
        // Begin Setting Region
        // This settings are for Avery Label number 5160. Modify the settings below based on your label number

        // Set page dimensions
        static int topMargin = 36;
        static int bottomMargin = 36;
        static float rightMargin = 13.5f;
        static float leftMargin = 13.5f;

        // Set the number of labels per page
        static int maximumColumns = 3;
        static int maximumRows = 10;

        // Set the spacing between the labels
        static int horizontalSpace = 9;
        static int verticalSpace = 0;

        // These margins are on the labels themsleves
        static int labelTopBottomMargin = 5;
        static int labelLeftRightMargin = 15;

        // End Setting Region
        //////////////////////////////////////////////////

        static Document document;
        static Page page;
        static float currentColumn, currentRow, labelWidth, labelHeight;
        static string companyName;
        static string contactName;
        static string phone;
        static string fax;

        public static List<MailingLabelsData.Customer> mailingLabelsData = MailingLabelsData.Customers;


        public static void Run(string outputPdfPath)
        {
            // Put user code to initialize the page here
            document = new Document
            {
                Creator = "MailingLabels.aspx",
                Author = "ceTe Software",
                Title = "Mailing Labels"
            };
            page = new Page(PageSize.Letter, PageOrientation.Portrait);

            // Entrypoint for the labels
            currentRow = 1;
            currentColumn = 1;

            foreach (var data in mailingLabelsData)
            {
                companyName = SafeDataNull(data.CompanyName);
                contactName = SafeDataNull(data.ContactName);
                phone = SafeDataNull(data.Phone);
                fax = SafeDataNull(data.Fax);
                AddLabel();
            }
            if (page.Elements.Count > 0)
                document.Pages.Add(page);

            // Outputs the document to the current web page
            document.Draw(outputPdfPath);
        }

        public static float FindLabelHeight()
        {
            return (page.Dimensions.Height - (page.Dimensions.TopMargin + page.Dimensions.BottomMargin) - ((maximumRows - 1) * verticalSpace)) / maximumRows;
        }

        public static float FindLabelWidth()
        {
            return (page.Dimensions.Width - (page.Dimensions.RightMargin + page.Dimensions.LeftMargin) - ((maximumColumns - 1) * horizontalSpace)) / maximumColumns;
        }

        public static void AddLabel()
        {
            // Add a new page if you are beyond the maximum Rows
            if (currentRow == maximumRows + 1)
            {
                document.Pages.Add(page);
                currentRow = 1;
            }

            // Determines if the the label belongs in the first row or first column of the page
            if (currentColumn > 1 & currentRow > 1)
            {
                AddToPage();
            }
            else if (currentColumn > 1 & currentRow == 1)
            {
                AddToFirstRow();
            }
            else if (currentColumn == 1 & currentRow > 1)
            {
                AddToFirstColumn();
            }
            else
            {
                page = new Page(PageSize.Letter, PageOrientation.Portrait);
                page.Dimensions.TopMargin = topMargin;
                page.Dimensions.BottomMargin = bottomMargin;
                page.Dimensions.RightMargin = rightMargin;
                page.Dimensions.LeftMargin = leftMargin;
                labelWidth = FindLabelWidth();
                labelHeight = FindLabelHeight();
                AddToFirstRowColumn();
            }

            // Incremment your row if you are beyond the maximum columns
            if (currentColumn == maximumColumns + 1)
            {
                currentRow = currentRow + 1;
                currentColumn = 1;
            }
        }

        // Adds the label on at least row 2 column 2 of the page
        public static void AddToPage()
        {
            float x;
            float y;
            y = (currentRow - 1) * (labelHeight + verticalSpace);
            x = (currentColumn - 1) * (labelWidth + horizontalSpace);
            AddLabelInfo(x, y);
            currentColumn = currentColumn + 1;
        }

        // Adds the label on the first row of labels
        public static void AddToFirstRow()
        {
            float x;
            float y;
            y = 0;
            x = (currentColumn - 1) * (labelWidth + horizontalSpace);
            AddLabelInfo(x, y);
            currentColumn = currentColumn + 1;
        }

        // Adds the label to the First column of labels
        public static void AddToFirstColumn()
        {
            float x;
            float y;
            y = (currentRow - 1) * (labelHeight + verticalSpace);
            x = 0;
            AddLabelInfo(x, y);
            currentColumn = currentColumn + 1;
        }

        // Adds only the first label of every page (row 1 column 1)
        public static void AddToFirstRowColumn()
        {
            float x;
            float y;
            y = 0;
            x = 0;
            AddLabelInfo(x, y);
            currentColumn = currentColumn + 1;
        }

        // This is where you format the look of each label
        public static void AddLabelInfo(float x, float y)
        {
            TextArea txt1 = new TextArea(companyName, x + labelLeftRightMargin, y + labelTopBottomMargin, labelWidth - (labelLeftRightMargin * 2), 11, Font.TimesRoman, 11);
            TextArea txt2 = new TextArea(contactName, x + labelLeftRightMargin, y + labelTopBottomMargin + 12, labelWidth - (labelLeftRightMargin * 2), 11, Font.TimesRoman, 11);
            TextArea txt3 = new TextArea(phone, x + labelLeftRightMargin, y + labelTopBottomMargin + 24, labelWidth - (labelLeftRightMargin * 2), 11, Font.TimesRoman, 11);
            TextArea txt4 = new TextArea(fax, x + labelLeftRightMargin, y + labelTopBottomMargin + 36, labelWidth - (labelLeftRightMargin * 2), 11, Font.TimesRoman, 11);
            page.Elements.Add(txt1);
            page.Elements.Add(txt2);
            page.Elements.Add(txt3);
            page.Elements.Add(txt4);
        }

        public static string SafeDataNull(string value)
        {
            if (value == null)
                return String.Empty;
            else
                return value;
        }
    }
}

File: coresuite-dotnet-core-console-vb/generator/MailingLabels.vb
Imports System.Convert
Imports System.Data.SqlClient
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.PageElements
Imports Microsoft.Extensions.Configuration

Public Class MailingLabels
    '/////////////////////////////////////////////////
    ' Begin Setting Region
    ' This settings are for Avery Label number 5160. Modify the settings below based on your label number
    ' Set page dimensions
    Shared TopMargin As String = 36
    Shared BottomMargin As String = 36
    Shared RightMargin As String = 13.5
    Shared LeftMargin As String = 13.5

    'Set the number of labels per page
    Shared MaximumColumns As String = 3
    Shared MaximumRows As String = 10

    'Set the spacing between the labels
    Shared HorizontalSpace As String = 9
    Shared VerticalSpace As String = 0

    'These margins are on the labels themsleves
    Shared LabelTopBottomMargin As String = 5
    Shared LabelLeftRightMargin As String = 15

    ' End Setting Region
    '/////////////////////////////////////////////////

    'Dim the page level variables
    Shared MyDocument As ceTe.DynamicPDF.Document
    Shared MyPage As Page
    Shared CurrentColumn As Integer
    Shared CurrentRow As Integer
    Shared LabelWidth As Single
    Shared LabelHeight As Single
    Shared CompanyName As String
    Shared ContactName As String
    Shared Phone As String
    Shared Fax As String

    Public Shared mailingLablesData As List(Of MailingLabelsData.Customer) = MailingLabelsData.Customers

    Public Shared Sub Run(outputPdfPath As String)
        'Put user code to initialize the page here
        'Create Document object and set properties

        MyDocument = New ceTe.DynamicPDF.Document
        MyPage = New Page(PageSize.Letter, PageOrientation.Portrait)

        MyDocument.Creator = "MailingLabels.aspx"
        MyDocument.Author = "ceTe Software"
        MyDocument.Title = "Mailing Labels"

        'Entrypoint for the labels
        Dim CurrentRow = 1
        Dim CurrentColumn = 1

        'Loop over the RecordSet and add each label 
        For Each data As MailingLabelsData.Customer In mailingLablesData
            CompanyName = SafeDataNull(data.CompanyName)
            ContactName = SafeDataNull(data.ContactName)
            Phone = SafeDataNull(data.Phone)
            Fax = SafeDataNull(data.Fax)

            AddLabel()
        Next

        If MyPage.Elements.Count > 0 Then
            MyDocument.Pages.Add(MyPage)
        End If

        'Outputs the document to the current web page
        MyDocument.Draw(outputPdfPath)

    End Sub

    Public Shared Function FindLabelHeight() As Single

        FindLabelHeight = (MyPage.Dimensions.Height - (MyPage.Dimensions.TopMargin + MyPage.Dimensions.BottomMargin) - ((MaximumRows - 1) * VerticalSpace)) / MaximumRows

    End Function

    Public Shared Function FindLabelWidth() As Single

        FindLabelWidth = (MyPage.Dimensions.Width - (MyPage.Dimensions.RightMargin + MyPage.Dimensions.LeftMargin) - ((MaximumColumns - 1) * HorizontalSpace)) / MaximumColumns

    End Function

    Public Shared Sub AddLabel()

        'Add a new page if you are beyond the maximum Rows
        If CurrentRow = MaximumRows + 1 Then

            MyDocument.Pages.Add(MyPage)

            CurrentRow = 1
        End If

        'Determines if the the label belongs in the first row or first column of the page
        If CurrentColumn > 1 And CurrentRow > 1 Then
            AddToPage()
        ElseIf CurrentColumn > 1 And CurrentRow = 1 Then
            AddToFirstRow()
        ElseIf CurrentColumn = 1 And CurrentRow > 1 Then
            AddToFirstColumn()
        Else
            MyPage = New Page(PageSize.Letter, PageOrientation.Portrait)

            MyPage.Dimensions.TopMargin = TopMargin
            MyPage.Dimensions.BottomMargin = BottomMargin
            MyPage.Dimensions.RightMargin = RightMargin
            MyPage.Dimensions.LeftMargin = LeftMargin

            'Calculates how big the Width and height of each label should be according to the margins and dimensions provided above
            LabelWidth = FindLabelWidth()
            LabelHeight = FindLabelHeight()

            AddToFirstRowColumn()
        End If

        'Incremment your row if you are beyond the maximum columns
        If CurrentColumn = MaximumColumns + 1 Then
            CurrentRow = CurrentRow + 1
            CurrentColumn = 1
        End If
    End Sub

    'Adds the label on at least row 2 column 2 of the page
    Public Shared Sub AddToPage()
        Dim X, Y As Single
        Y = (CurrentRow - 1) * (LabelHeight + VerticalSpace)
        X = (CurrentColumn - 1) * (LabelWidth + HorizontalSpace)
        AddLabelInfo(X, Y)
        CurrentColumn = CurrentColumn + 1
    End Sub

    'Adds the label on the first row of labels
    Public Shared Sub AddToFirstRow()
        Dim X, Y As Single
        Y = 0
        X = (CurrentColumn - 1) * (LabelWidth + HorizontalSpace)
        AddLabelInfo(X, Y)
        CurrentColumn = CurrentColumn + 1
    End Sub

    'Adds the label to the First column of labels
    Public Shared Sub AddToFirstColumn()
        Dim X, Y As Single
        Y = (CurrentRow - 1) * (LabelHeight + VerticalSpace)
        X = 0
        AddLabelInfo(X, Y)
        CurrentColumn = CurrentColumn + 1
    End Sub

    'Adds only the first label of every page (row 1 column 1)
    Public Shared Sub AddToFirstRowColumn()
        Dim X, Y As Single
        Y = 0
        X = 0
        AddLabelInfo(X, Y)
        CurrentColumn = CurrentColumn + 1
    End Sub

    'This is where you format the look of each label
    Public Shared Sub AddLabelInfo(ByVal X As Single, ByVal Y As Single)

        Dim Txt1 As TextArea = New TextArea(CompanyName, X + LabelLeftRightMargin, Y + LabelTopBottomMargin, LabelWidth - (LabelLeftRightMargin * 2), 11, Font.TimesRoman, 11)
        Dim Txt2 As TextArea = New TextArea(ContactName, X + LabelLeftRightMargin, Y + LabelTopBottomMargin + 12, LabelWidth - (LabelLeftRightMargin * 2), 11, Font.TimesRoman, 11)
        Dim Txt3 As TextArea = New TextArea(Phone, X + LabelLeftRightMargin, Y + LabelTopBottomMargin + 24, LabelWidth - (LabelLeftRightMargin * 2), 11, Font.TimesRoman, 11)
        Dim Txt4 As TextArea = New TextArea(Fax, X + LabelLeftRightMargin, Y + LabelTopBottomMargin + 36, LabelWidth - (LabelLeftRightMargin * 2), 11, Font.TimesRoman, 11)

        MyPage.Elements.Add(Txt1)
        MyPage.Elements.Add(Txt2)
        MyPage.Elements.Add(Txt3)
        MyPage.Elements.Add(Txt4)

    End Sub

    Shared Function SafeDataNull(ByVal Value As String) As String
        If Value Is Nothing Then
            Return String.Empty
        Else
            Return Value
        End If
    End Function
End Class

